src/pamiq_core/console/cui.py

```diff
@@ -1,84 +1,109 @@
 import argparse
-import cmd
 import json
-from typing import override
 
 import httpx
+from prompt_toolkit import prompt
+from prompt_toolkit.completion import WordCompleter
 
 
-class Console(cmd.Cmd):
+class Console:
     """pamiq-console.
 
     Users can Control pamiq with CUI interface interactively.
     """
 
-    intro = 'Welcome to the PAMIQ console. "help" lists commands.\n'
-    prompt: str
+    status: str
 
     def __init__(self, host: str, port: int) -> None:
         """Initialize CUI interface."""
         super().__init__()
         self._host = host
         self._port = port
-        self._fetch_status()
+        self._completer = WordCompleter(self.get_all_commands())
+        self.fetch_status()
 
-    @override
-    def onecmd(self, line: str) -> bool:
+    def fetch_status(self) -> None:
+        """Check WebAPI status."""
+        try:
+            response = httpx.get(f"http://{self._host}:{self._port}/api/status")
+        except httpx.RequestError:
+            self.status = "offline"
+            return
+        self.status = json.loads(response.text)["status"]
+
+    def get_all_commands(self) -> list[str]:
```

Comment:
ここコマンドが増えてくると重くなると思うので、 `functool.cached_property`を使うと良いと思う。
```suggestion
    @cached_property
    def all_commands(self) -> list[str]:
```

Reply:
https://github.com/MLShukai/pamiq-core/pull/217/commits/936716955ad7db239fd645660655258346ff9d9c で追加しといた